### ISSUE TITLE
[ERROR] sudo: unable to send audit message on Synology

### ISSUE THREAD

**Issue by heseifert** [pi-hole/docker-pi-hole#243]
Hello,
i got this error on my docker logs.

`sudo: unable to send audit message: Operation not permitted`

If i do anything on the WebUI i got this error.

My Setup:
- Synology DSM 6.1.5 DS1515+
- Lastest docker-pi-hole Image

Can any help me?

**Comment by diginc** [comment]
I would start with synology + that error message unless this is very specific to my container.

have you looked at these posts?  https://duckduckgo.com/?q=site%3Aforum.synology.com+Operation+not+permitted&t=ffab&ia=web

Do you use volumes and if you remove them and re-create the container does this error go away?

**Comment by jcacavias** [comment]
I'm getting a similar error message each time I try to press the enable button.  Enable appears to work, but as soon as I refresh the page, the enable button is back.  Otherwise, pi-hole appears to be working.  Here are the errors that appear in the logs...

sudo: unable to send audit message: Unknown error -1
sudo: pam_open_session: System error | stdout
sudo: policy plugin failed session initialization

**Comment by edudojevic** [comment]
Same here when I try `sudo pihole -r`:
```
root@diginc-pi-hole1:/# sudo pihole -r
sudo: unable to send audit message: Unknown error -1
sudo: pam_open_session: System error
sudo: policy plugin failed session initialization
```

Also having the same problems through the Web UI, where I cannot save anything. It seems like a permissions issue, but since I'm using the already built diginc pihole docker image I would expect this to work out of the box. Running this on Synology DS916+.

Any help is appreciated.

**Comment by ShagoY** [comment]
Hi, i got this error only with --net=host (the only way to see "Top Clients ?")

**Comment by kn0rki** [comment]
Hi folks,
same errors for me on a DS412+.

@ShagoY the --net=host works without NAT. You can add a new network and disable ip masquerading to enable the top clients.

**Comment by ShagoY** [comment]
Hi @kn0rki, i just try in docker but no result, i think i need more information, have you a guide ?

**Comment by Tearow** [comment]
Same issue.
When ever i try to add something in whitelist or add something in blacklist or any setting i receive in docker terminal.

Sudo: Unable to send audit message: Unknown error -1
Sudo: pam_open_session: System error
Sudo: policy plugin failed session initialization

did somebody solve it already. i saw already multiple issues about this but still without solution.

Using DS412+ with docker and Diginc/pi-hole:latest

**Comment by will-code-for-pizza** [comment]
Using DS 2413+ with docker and Diginc/pi-hole:latest
Same issue here.

docker logs gave me:

```
sudo: unable to send audit message: Unknown error -1
sudo: pam_open_session: System error
sudo: policy plugin failed session initialization
```
Additionally:

`dnsmasq: failed to load names from /etc/pihole/black.list: No such file or directory`

Status in web UI changes from Active to Unknown a few seconds after clicking "Enable".

**Comment by doucheymcdoucherson** [comment]
I have the same problem. Mapping the black.list to a file doesn't fix it.

**Comment by florid2** [comment]
same issue here. And the admin console shows status unknown.

**Comment by florid2** [comment]
I tried to isolate the problem whether an image issue or Synology DSM restriction. 
I only have a spare raspberry pi 3b to play with. I pulled diginc/pi-hole-multiarch:debian_armhf on HypriotOS docker environment. The same issue was not discovered in that setup although version number not displayed. 
Right now what I can do is to use Raspberry Pi to service Pi-Hole directly on Raspbian OS which works perfectly. I do hope this issue can be fixed soon so that I can just use one NAS.

**Comment by doucheymcdoucherson** [comment]
I ended up using virtual station to spin up a debian vm which I run pihole on.

**Comment by florid2** [comment]
@doucheymcdoucherson thanks for the suggestion, I tried virtual station but not really satisfied with the resources it consumed. I do not run an extremely powerful synology nas. 
And considering NAS requires regularly software upgrade, 10 minutes DNS downtime is not acceptable even for a home network. 
So I am now using a spare Raspberry Pi directly. All issue fixed.

**Comment by CrypticOne** [comment]
I have issue 286. I think they probably could be merged. I know the webinterface uses superuser commands to update the config files. 

My docker image reports the following when I try to use sudo:

root@pihole:/# sudo                                                                                                                          
sudo: unable to resolve host pihole 

root@pihole:/# sudo date                                                                                                                     
sudo: unable to resolve host pihole                                                                                                          
sudo: unable to send audit message: Unknown error -1                                                                                         
sudo: pam_open_session: System error                                                                                                         
sudo: policy plugin failed session initialization

**Comment by Ulisus** [comment]
So i am having the same problems with the latest build on a Synology NAS. So i tried installing the "debian_v3.2.1" branch.

Now the settings are finally saved!

The only problems here are:
- Version numbers not showing in the right bottom corner
- Opening "Query Log" produces a popup "DataTables warning: table id=all-queries - Requested unknown parameter '5' "
- I still get the sudo: unable to send audit message: Operation not permitted" in the logs.


(By the way debian_3.1.4 didn't work, because the webport couldn't be changed from 80)

**Comment by McMac** [comment]
Running into the same issue with the System error/Unknown error -1.

I managed to drill down to it:
```bash
docker exec -it pi-hole /bin/bash
root@pi-hole:/# su
su: System error
root@pi-hole:/# sudo -i
sudo: unable to send audit message: Unknown error -1
sudo: pam_open_session: System error
sudo: policy plugin failed session initialization
root@pi-hole:/#
```

After some hefty googling and testing this issue only seem to occur when running with --net=host not with --net=bridge. There is a thread [https://github.com/moby/moby/issues/5899] for another container describing the same issue and the solution seems to be "Install libpam with --disable-audit option on Ubuntu 14.04"

**Comment by diginc** [comment]
@McMac Thanks for digging into this.  Looking at that Docker Moby ticket, it sounds like it's an older kernel problem rather than a docker problem.  Can you confirm what docker and kernel versions you have with these commands?

```
docker version
docker info
```

My oldest test system I have is 3.19 (Vivid) and I've never seen this problem, so given the newest kernel mentioned in that thread is 3.17 I'd think somewhere between those two the problem was resolved.   I'd suggest looking into any upgrade options Synology has for kernel and docker versions.

**Comment by diginc** [comment]
Yup "Kernel Version: 3.10.102" :( 

Not much I can do here for this.  I searched a little about what synology allows upgrades for but it depends on the the line/model.

**Comment by R91g** [comment]
I think that DSM 6.2 has Kernel 4.4
Maybe with this version Pihole works fine.

Xpenology with loader 1.03b now works with DSM 6.2-23739 update 2:
https://mega.nz/#F!ZlkHQTTb!keje3RK017OjTp3vuWb-Cw

**Comment by ituri** [comment]
> I think that DSM 6.2 has Kernel 4.4
No, it doesn't. My DS916+ runs on the latest DSM 6.2-23739 Update 2 and it comes with Kernel 3.10.105.

**Comment by Anatharias** [comment]
So I used a brand new DSM 6.2 on a DS1513+ (never used as a Docker host before)
launched the new docker pihole/pihole v4 image
and once setup is complete

the same message appears each time I want to enable the service:

System error                                                                               
sudo: unable to resolve host pihole.n8                                                     
sudo: unable to send audit message: Unknown error -1                                       
sudo: pam_open_session: System error                                                       
sudo: policy plugin failed session initialization

**Comment by McMac** [comment]
I am experiencing the same with v4.

I created a ticket with Synology Support (#2061260) with the current details also referring to this thread. 

In the meantime I'll take a shot at the --disable-audit idea.

**Comment by McMac** [comment]
Ok, I tried to rebuild everything by creating a Debian image with the —disable-audit flag, then took @diginc’s sources and made an amd64_disable-audit dockerfile and modified his Dockerfile.py to use my modified one however I must have made a snafu somewhere and I somehow ended up with the same error. It was a long shot to simply replace Debian “Jessie” with “Stretch” and hope for it to work. Just running the plain Debian image and doing an “su” in it gave me a System error so no dice there. Maybe someone more skilled knows how to get it working.

**To anyone using it on Synology: which one was the latest version which worked for you?**

(Want to switch back as currently dns resolution is sketchy, saves don’t work, “enable” is not sticky, can’t save settings nor whitelist urls via the UI”. It also no longer takes the machine names from my router - AVM Fritz!Box 7490 - however the names are used properly in the statistics however the DNS part doesn’t seem to resolve the names?!?) liked the low resource consumption but this is not useable unfortunately.
And Synology: “does not support custom docker images” - of course, but using an outdated Linux kernel is starting to break Docker support. Let’s see how much relaying the details provided in my ticket is actually going to move anything. 
So much on updates from my side and to anyone else finding this thread in the search or an answer...

**Comment by Anatharias** [comment]
Like I wrote on my previous edited comment, it worked with specified ports ie 80 to 8888, etc
For the sake of testing kept 53 and 67 to their own ports 

and Pihole is running fine 
![image](https://user-images.githubusercontent.com/17396591/43998193-0033e3f2-9dbe-11e8-8799-e44cdfc2da5d.png)

![image](https://user-images.githubusercontent.com/17396591/43998215-4c367652-9dbe-11e8-92ea-1ddcbdd3a054.png)

**Comment by McMac** [comment]
Thanks for the reminder @Anatharias that everything works when using --net=bridge. 👍 
One loses the details about the source of any DNS request (-> stats, issue tracking) but on second thought do I really need those details?! Probably not. 
It's the "right-Docker-way" anyways. :)

**Comment by Xento** [comment]
I think I fixed it using this commands:
https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile

**Comment by hummel124** [comment]
> I think I fixed it using this commands:
> https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile

Confirmed! This did the trick :)
I'm running the Docker Container on my Synology DS1812 with DSM 6.2-23739 Update 2. Docker version ist 17.05.0-0379

Before executing the commands I had to modify the /etc/apt/sources.list: (Lines with deb-src were missing)
```
deb http://deb.debian.org/debian stretch main
deb-src http://deb.debian.org/debian stretch main
deb http://deb.debian.org/debian-security/ stretch/updates main
deb-src http://deb.debian.org/debian-security/ stretch/updates main
deb http://deb.debian.org/debian stretch-updates main
deb-src http://deb.debian.org/debian stretch-updates main
```


Thanks
Chris

**Comment by Xento** [comment]
Yeah I had to add this lines, too.
I forgot to mention that ;-)

**Comment by Azzhoe** [comment]
> > I think I fixed it using this commands:
> > https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile
> 
> Confirmed! This did the trick :)
> I'm running the Docker Container on my Synology DS1812 with DSM 6.2-23739 Update 2. Docker version ist 17.05.0-0379
> 
> Before executing the commands I had to modify the /etc/apt/sources.list: (Lines with deb-src were missing)
> 
> ```
> deb http://deb.debian.org/debian stretch main
> deb-src http://deb.debian.org/debian stretch main
> deb http://deb.debian.org/debian-security/ stretch/updates main
> deb-src http://deb.debian.org/debian-security/ stretch/updates main
> deb http://deb.debian.org/debian stretch-updates main
> deb-src http://deb.debian.org/debian stretch-updates main
> ```
> Thanks
> Chris

Hi, I am also running pi-hole on my Synology 1512 NAS running DSM 6.2. However my pi-hole is behaving quite strange. I cannot seem to save any whitelist/blacklist entries, cannot save any of my settings, the version does not show on the bottom right, instead it shows N/A. When I hit enable in the dashboard it wlil go back to the yellow circle status and won't stay on, but it is somehow still blocking. I am running the official pi-hole.

Could you please show me what you did to get this to work without any problems?

**Comment by diginc** [comment]
Please create a new issue, sounds like you have a different problem. the
yellow circle problem was also a old bug make sure your images up to date.

**Comment by Azzhoe** [comment]
I'm not using the pi-hole from diginc but the official one. does this matter?

**Comment by diginc** [comment]
make sure you're using [pihole/pihole](https://hub.docker.com/r/pihole/pihole/)

**Comment by alek-05** [comment]
> > I think I fixed it using this commands:
> > https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile
> 
> Confirmed! This did the trick :)
> I'm running the Docker Container on my Synology DS1812 with DSM 6.2-23739 Update 2. Docker version ist 17.05.0-0379
> 
> Before executing the commands I had to modify the /etc/apt/sources.list: (Lines with deb-src were missing)
> 
> ```
> deb http://deb.debian.org/debian stretch main
> deb-src http://deb.debian.org/debian stretch main
> deb http://deb.debian.org/debian-security/ stretch/updates main
> deb-src http://deb.debian.org/debian-security/ stretch/updates main
> deb http://deb.debian.org/debian stretch-updates main
> deb-src http://deb.debian.org/debian stretch-updates main
> ```
> Thanks
> Chris

I tried but I receive this error message when I run the last command:

>dpkg-buildpackage: info: source package pam
dpkg-buildpackage: info: source version 1.1.8-3.6
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Adrian Bunk <bunk@debian.org>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build pam-1.1.8
 debian/rules clean
dh clean --with quilt,autoreconf
   dh_testdir
   dh_auto_clean
   dh_quilt_unpatch
Patch make_documentation_reproducible.patch does not remove cleanly (refresh it or enforce with -f)
dh_quilt_unpatch: quilt --quiltrc /dev/null pop -a || test $? = 2 returned exit code 1
debian/rules:21: recipe for target 'clean' failed
make: *** [clean] Error 25
dpkg-buildpackage: error: debian/rules clean gave error exit status 2
E: Build command 'cd pam-1.1.8 && dpkg-buildpackage -b -uc' failed.

I'm not an expert, could you please help me understand how to proceed?

Thank you.

**Comment by Azzhoe** [comment]
I am using that one. I’m going to redownload it again. I chose the ‘latest’ one from the drop down box, perhaps I’m choosing the wrong build? I am on a Synology 1512+ NAS running I believe on Intel Atom Processor. So Do I get the latest version or something else?

The menu shows the following:

Latest
4.0.0.-1
4.0.0.-1_aarch64
4.0.0.-1_amd64
4.0.0.-1_armhf
development

**Comment by diginc** [comment]
Great find @Xento, seems to be working for everyone.  I'll work on getting this implemented in development branch now.  

> #Docker PAM workaround
> 
> If you start your docker container with host networking enabled then due to a kernel bug commands like `su` does not work.
> One solution for this issue is to patch the host kernel with the following: [AUDIT: Allow login in non-init namespaces](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/patch/?id=543bc6a1a987672b79d6ebe8e2ab10471d8f1047
> )
> 
> 
> Another option is to patch the libpam on the guest OS running inside the docker container. For this purpose the `sequenceiq/pam` images are created for the most popular operating systems.

**Comment by diginc** [comment]
Give pihole/pihole:development a try and see if it removes the audit errors.

**Comment by diginc** [comment]
@Azzhoe If you're seeing the 'sudo: unable to send audit message' error message please try the image `pihole/pihole:development`.  I just the fix being discussed to that image last night.

If you're seeing a different issue please fill out a new issue with the full template details.  Synology is only community supported, I don't have hardware to reproduce issues or write any tutorials, but we'll try our best to figure out your problems if you provide enough data.

**Comment by mylala** [comment]
@diginc: Hey, thanks having a look into the issue, again! 

I checked the development version and get the follow error msgs:
sudo: unable to send audit message: Unknown error -1                                                                                                    
sudo: pam_open_session: System error                                                                                                                    
sudo: policy plugin failed session initialization

**Comment by mizhgan** [comment]
> sudo: unable to send audit message: Unknown error -1

> sudo: pam_open_session: System error

> sudo: policy plugin failed session initialization

Hello. I got the same error as above. Using development image.
Web interface is not saving changes again.

**Comment by diginc** [comment]
Hmmm, I need a way to verify the PAM --disable-audit install happened
correctly.  I pretty much just copied the install of pam out of
https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile
into the install.  The readme of sequenceiq/docker-pam also says this
command can be used to verify the issue is fixed :

`sudo docker run -it --rm --net=host --entrypoint=bash
pihole/pihole:development -c 'useradd testuser; su testuser;'`

(no error should return, `id` should say you are testuser)

**Comment by Azzhoe** [comment]
I’m getting similar errors like everyone else when checking the log. I have the latest pihole/pihole:developer build installed. This is what my log says:

  [i] Consolidating blocklists...
  [✓] Consolidating blocklists
stdout
01:09:56
  [i] Extracting domains from blocklists...
  [✓] Extracting domains from blocklists
stdout
01:09:56
  [i] Number of domains being pulled in by gravity: 150449
stdout
01:09:56
  [i] Removing duplicate domains...
  [✓] Removing duplicate domains
stdout
01:09:56
  [i] Number of unique domains trapped in the Event Horizon: 127494
stdout
01:09:56
  [i] Nothing to whitelist!
stdout
01:09:56
  [i] Number of regex filters: 0
stdout
01:09:56
  [i] Parsing domains into hosts format...
  [✓] Parsing domains into hosts format
stdout
01:09:56
  [i] Cleaning up stray matter...
  [✓] Cleaning up stray matter
stdout
01:09:56
stdout
01:09:56
  [i] Force-reloading DNS service...
  [✓] Force-reloading DNS service
stdout
01:09:56
  [✓] DNS service is running
stdout
01:09:56
  [✓] Pi-hole blocking is Enabled
stdout
01:09:56
[cont-init.d] 20-start.sh: exited 0.
stdout
01:09:56
[cont-init.d] done.
stdout
01:09:56
[services.d] starting services
stdout
01:09:56
Starting pihole-FTL (no-daemon)
stdout
01:09:56
Starting crond
stdout
01:09:56
Starting lighttpd
stdout
01:09:56
[services.d] done.
stdout
01:10:01
stdout
01:10:01
System error
stdout
01:10:19
sudo: unable to resolve host pihole-pihole4
stdout
01:10:19
sudo: unable to send audit message: Unknown error -1
stdout
01:10:19
sudo: pam_open_session: System error
stdout
01:10:19
sudo: policy plugin failed session initialization
stdout
01:10:22
sudo: unable to resolve host pihole-pihole4
stdout
01:10:22
sudo: unable to send audit message: Unknown error -1
stdout
01:10:22
sudo: pam_open_session: System error
stdout
01:10:22
sudo: policy plugin failed session initialization
stdout
01:10:24
sudo: unable to resolve host pihole-pihole4
stdout
01:10:24
sudo: unable to send audit message: Unknown error -1
stdout
01:10:24
sudo: pam_open_session: System error
stdout
01:10:24
sudo: policy plugin failed session initialization

**Comment by mizhgan** [comment]
@diginc that`s what i get after i use this command:

> admin@NAS:~$ sudo docker run -it --rm --net=host --entrypoint=bash pihole/pihole:development -c 'useradd testuser; su testuser;'
> su: System error

**Comment by gazzer82** [comment]
Ditto, same problem here, with the latest dev build.

**Comment by diginc** [comment]
@mizhgan is right, most likely because I forgot the deb-src sources.  Will try again tonight, surprised the build didn't straight up fail.

**Comment by diginc** [comment]
I pushed `pihole/pihole:synology` [tag](https://hub.docker.com/r/pihole/pihole/tags/) last night based off the [synology](branch).  Give it a try and let me know if it works, it definitely got the right change this time.  

Some notes/concerns after getting things working last night:

- The size of the image grew by ~40%.  I'd like to reduce that back down before there's chance of including this in the main image.
- Build times grew  more than 40%, it took what felt like an hour for all 4 image flavors.  Non amd64 builds suffer the most, I bring this up because synology only requires an amd64 build, correct?  A fork is a possibility if I can't fix these above 2 issues.
  - A custom base image is a potential solution
- Some of the automated tests were failing, I don't think they really failing, these particular IPv6 have grown flaky lately due to disk sync delay problems I believe.

**Comment by Azzhoe** [comment]
@diginc Thanks, so before I redownload this all over again on my Synology NAS, could you please specify which exact build I need? I'm referring to the pull down menu where it gives you options like:

latest
development
amd64
deb
etc

I want to be sure I'm correctly choosing the right package for my Synology. I have a 1512+ Model that runs on Intel Atom Processor (64 bit I believe). So will this new version finally fix the annoying version not displaying and the ability to write/save my settings such as the whitelist/blacklist, the yellow status circle not switching to green, etc?

**Comment by mizhgan** [comment]
It is still makes this errors in console
`sudo: unable to send audit message: Unknown error -1`
but it definitely works now, on synology tagged image.
Web interface is working, settings are working.

**Comment by diginc** [comment]
@Azzhoe synology regular tag.  The architecture tags are never needed by users anymore.  The non-architecture tags will auto-detect your architecture and use the appropriate tag.  It acts as a magic alias basically.

**Comment by Azzhoe** [comment]
@diginc Thank you! I will give it a try and report back to you! Thanks so much for doing this!

**Comment by Xento** [comment]
I think it grew so much because of the source files and the programms to compile them.

**Comment by diginc** [comment]
Glad to hear it's working @mizhgan 

> I think it grew so much because of the source files and the programms to compile them.

I'll be sure to add `apt autoclean` and search for & remove any build essentials apps to see if that helps.  I made sure to delete the [debs](https://github.com/pi-hole/docker-pi-hole/pull/358/files#diff-6aa367da3fdfc952b15def7759aa5f0bR20) created for pam thinking that'd get most of it but ran out of time last night so just pushed a WIP branch image for testing.
 
#358 is available for review if you can think of any other cleanup command ideas please feel free to add a comment there.

**Comment by Azzhoe** [comment]
@diginc I can report that this version finally is working and has fixed all those problems I had! But when looking at the log, I still see the following errors, however it doesn't seem to affect anything from what I'm aware of.

sudo: unable to resolve host pihole-pihole5
sudo: unable to send audit message: Unknown error -1

**Comment by diginc** [comment]
I've deleted the temporary test `synology` tag and moved the fix onto the `development` tag.  As mentioned [in this comment](https://github.com/pi-hole/docker-pi-hole/pull/358#issuecomment-434907739) - I split out the pam fixes into a new base image to reduce build time and proper cleanup of the image got it back in-line with the `latest` tag's size.

**Comment by Azzhoe** [comment]
@diginc I've just re-downloaded the latest pihole/pihole:development version it came out to be about 345MB vs 485MB from the pihole/pihole:synology version. But now, I am experiencing all the same problems as before. I can't enable it without it returning back to the yellow status light, and I can't save any settings.

**Comment by diginc** [comment]
Hmm sorry about that, something must have gone wrong with the build.  I thought the size was a little too close to latest - it should be probably ~20Mb larger

**Comment by Azzhoe** [comment]
@diginc just downloaded the development build and it's still the same 345MB.

**Comment by Azzhoe** [comment]
@diginc any idea when this the development build will be updated, it’s still showing as 345mb.

**Comment by diginc** [comment]
Not quite sure why you're getting the same error, I'm pretty sure the pam fix should've been included in what I pushed.  Size being reduced is expected because I optimized it.  Has anyone else tested `development` to confirm if it has the fix in it or not?  

When you say you have the issue again do you mean just the printed errors in the log or the actual failure of functionality?

Could you test the base image?

`sudo docker run -it --rm --net=host --entrypoint=bash pihole/debian-base -c 'useradd testuser; su testuser;'`
then run `id`

**Comment by Azzhoe** [comment]
@diginc I’m getting the exact same behaviors and errors as before. Printed errors and failure as in the status circle in pinhole dashboard is Yellow, and when I enable it, it goes back to being yellow, and my settings don’t save. Here’s a screenshot of the log

https://i.postimg.cc/SKj9BGw8/Screen-Shot-2018-11-03-at-8-36-12-PM.png <https://i.postimg.cc/SKj9BGw8/Screen-Shot-2018-11-03-at-8-36-12-PM.png>

Can you confirm that the pihole/pihole:development is 345mb? For now I’m reverting back to the pihole/pihole:synology build (487mb), that one works perfectly.

**Comment by dschaper** [comment]
There's something up between docker hub/cloud and actual deployment. As the image shows the latest push was 3 days ago for an image of 162 MB. But a `docker pull pihole/pihole:development` on a fresh install of docker pulls `pihole/pihole                      development         9a1363998a45        2 months ago        345MB`

![image](https://user-images.githubusercontent.com/2782850/47959121-9e158180-df99-11e8-8e29-344e864e5308.png)

**Comment by Azzhoe** [comment]
Strange, so how can I get the latest pihole/pihole:development? So the size of the latest version should be 162MB?

**Comment by dschaper** [comment]
That 162 MB is the size of the final layer. So the actual size on disk for the end user is the base image size plus that final layer. The dockerfile is a pretty clean build so I wouldn't expect it to be more than 300MB final size on disk. What concerns me is that the history (`docker history pihole/pihole:development`) shows the latest modifications to be 2 months ago while it was pushed 3 days ago. The history for latest seems to be fine however.

**Comment by Azzhoe** [comment]
I see, so is there anything that I can do on my end to get the latest version? Or do I need to wait a few days and it should update?

**Comment by dschaper** [comment]
I don't think there's anything to do on the user end. If I get a chance I'll try to build an image and push it to the hub or if diginc gets to it first. I don't think it's an automated build yet but we can check.

**Comment by Azzhoe** [comment]
Okay, thanks! In that case I’ll continue using the pihole/pihole:synology until the development build finally reaches the docker registry

**Comment by Azzhoe** [comment]
@diginc this one is working for me. It shows up as 290MB. Running just like it did previously. Thank you!

**Comment by jtdelia** [comment]
pinhole/pihole:dev worked for me as well, no more error messages and I am able to modify and save configurations, black/white lists.

**Comment by McMac** [comment]
Which one contains the latest fix for this issue? Still :dev or is it included in the latest 4.1.1?

**Comment by diginc** [comment]
They should both be fixed and good

**Comment by McMac** [comment]
Thanks @diginc. Unfortunately I seem to miss something. Kept my config since Pi-Hole 3 with the volumes and configs defined in http://tonylawrence.com/post/unix/synology/running-pihole-inside-docker/ also tried fresh setup with only basic setup taken ignoring my previous setupVars.conf.
- Still seeing "sudo: unable to send audit message: Unknown error -1" with 4.1.1 on Synology/Docker/host - is that supposed to be fixed or just not cause anymore troubles?!
- Query Log is empty despite "Show everything and record everything" being set and no exclusions set
- After original upgrade I got "DNS service is NOT running"
- Reverting back to a re-created pihole:v4.0_amd64, pointing to old volumes, pi-hole works again however Query Log still empty 

Had upgraded from 4.0 to 4.1.1. with the same config with only change of network=bridge to network=host. 

At least for me the docker way of keeping configs in volumes and updating binaries only isn't working. Anything I need to reset in between.

**Comment by doNmartinez** [comment]
I am running pihole in docker on my Synology's (4 atm: 2x ds1817+ and 2x 918+). Each of them is on the newest updates for DSM, Docker and Pi-Hole. After struggling around with getting pihole setup I finally managed to get it working. One of my Diskstations had Active Directory Controlle running for test purposes. If someone has a guide to run pihole on a Synology with activated AD that would be awesome.
Anyway I am getting the same Error Messages: sudo: unable to send audit message: Unknown error -1 on stdout. But the Webinterface seems to work fine.

**Comment by doNmartinez** [comment]
And dont get me wrong. I really appreciate the work you are doing. I am not complaining. I try to give you input since I really think pihole is an awesome solution for ad blocking. But atm the synology docker version is not running that smooth. And there are some essential guides missing.
For example which volume should be assigned to store settings and information during upgrade. Or how can I import a certificate for https:// with the docker image?

**Comment by DistractionRectangle** [comment]
@doNmartinez A previous comment linked to a guide for running pihole on synology, and the updated version seems rather thorough: http://tonylawrence.com/posts/unix/synology/free-your-synology-ports/

As for this issue, could you attach a bash shell to the pihole container and provide the output of: `apt-get update > /dev/null && apt list --installed | grep libpam`

**Comment by pralor** [comment]
This issue has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/synology-docker-error-unable-to-resolve-host-pi-hole-and-unable-to-send-audit-message-unknown-error-1/31691/2

**Comment by helmarw** [comment]
i ran into the same problem, is there a solution for it ?
installed the latest pihole docker 5.1.2 it is atm and configured it according to this howto: https://mariushosting.com/how-to-install-pi-hole-on-your-synology-nas/
sudo: unable to send audit message: Unknown error -1

and no statistics on the website
